pkg/scheduler/framework/runtime/mock_metrics_recorder.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package runtime
+
+import (
+	"sync"
+)
+
+// MockMetricsRecorder is a fake implementation of MetricsRecorder for testing.
+// It uses counters and stores call records for verification in tests.
+type MockMetricsRecorder struct {
+	mu sync.Mutex
```

Comment:
You can use `RWMutex` and use `RLock`, `RUnlock` in read-only places
```suggestion
	mu sync.RWMutex
```